course02/homework00/joost/vector.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <array>
+#include <stdexcept>
+
+class VectorInt
+{
+	public:
+		VectorInt():
+			mSize(0)
+		{}
+		
+		int size() const
+		{
+			return mSize;
+		}
+
+		int capacity() const
+		{
+			return mData.size();
```

Comment:
`mData.size()` returns a `std::size_t`, which is `unsigned`. In this particular case it is fine as your capacity is maximum 10, but otherwise this would not be entirely correct. The reason is that casting from `std::size_t` to `int` can lead to undefined behavior if the value is outside of the `int` range. As a solution, just return the correct type, i.e. `std::size_t`.